### PR TITLE
Fix Linux OS issues: #440 and #442

### PR DESCRIPTION
**Pull request by YoshiEnVerde**
#440:
Multiplied the network speed by 2^20 to normalize it from MiBs to Bytes.
Also added an overflow check (It shouldn't ever happen though, as that would mean network speeds above 7 Exabytes per second)

#442:
Instead of building a whole new method inside a util class to allow for the correct **/proc/[PID]/stat** file parsing (where the first two delimiters are " (" and ") ", instead of whitespaces), I just used the current file splitting, then concatenated any resulting token that belonged to the **name** field.

### CODE REVIEW

File: oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java [hunk 1]
```diff
@@ -226,9 +226,16 @@ public OSProcess getProcess(int pid) {
...
         // See man proc for how to parse /proc/[pid]/stat
-        proc.setName(split[1].replaceFirst("\\(", "").replace(")", ""));
+        int nameOffset = 0;
+        StringBuilder name = new StringBuilder(
+                split[1].replaceFirst("\\(", ""));
+        while (!split[1 + nameOffset].endsWith(")")) {
+            nameOffset++;
+            name.append(' ').append(split[1 + nameOffset]);
+        }
+        proc.setName(name.substring(0, name.length() - 1));
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java]: This all looks good, other than the part of me that thinks we should do some sort of error checking.  While we can probably rely on an actual `/proc/pid/stat` result to be formatted as required, we should check a "just in case" it's an empty string.  Perhaps test the existence of the `(` character in the string?

File: oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java [hunk 2]
```diff
@@ -249,14 +256,14 @@ public OSProcess getProcess(int pid) {
...
-        proc.setUserTime(ParseUtil.parseLongOrDefault(split[13], 0L) * 1000L / hz);
-        proc.setStartTime(bootTime + ParseUtil.parseLongOrDefault(split[21], 0L) * 1000L / hz);
+        proc.setParentProcessID(ParseUtil.parseIntOrDefault(split[3 + nameOffset], 0));
+        proc.setThreadCount(ParseUtil.parseIntOrDefault(split[19 + nameOffset], 0));
+        proc.setPriority(ParseUtil.parseIntOrDefault(split[17 + nameOffset], 0));
+        proc.setVirtualSize(ParseUtil.parseLongOrDefault(split[22 + nameOffset], 0L));
+        proc.setResidentSetSize(ParseUtil.parseLongOrDefault(split[23 + nameOffset], 0L) * this.memoryPageSize);
+        proc.setKernelTime(ParseUtil.parseLongOrDefault(split[14 + nameOffset], 0L) * 1000L / hz);
+        proc.setUserTime(ParseUtil.parseLongOrDefault(split[13 + nameOffset], 0L) * 1000L / hz);
+        proc.setStartTime(bootTime + ParseUtil.parseLongOrDefault(split[21 + nameOffset], 0L) * 1000L / hz);
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java]: similar to above comment, perhaps ensure `split` length exceeds `21 + nameOffset`

File: oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxNetworks.java [hunk 3]
```diff
@@ -52,6 +52,7 @@ public static void updateNetworkStats(NetworkIF netIF) {
         netIF.setPacketsRecv(FileUtil.getUnsignedLongFromFile(rxPacketsPath));
         netIF.setOutErrors(FileUtil.getUnsignedLongFromFile(txErrorsPath));
         netIF.setInErrors(FileUtil.getUnsignedLongFromFile(rxErrorsPath));
-        netIF.setSpeed(FileUtil.getUnsignedLongFromFile(speed));
+        long netSpeed = FileUtil.getUnsignedLongFromFile(speed) * 1024 * 1024;
+        netIF.setSpeed(netSpeed < 0 ?0 :netSpeed);
```
Comment (dbwiddis) [inline oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxNetworks.java]: can you run this through a code formatter?  Wouldn't ask if it was the only change, but I expect some spaces would be added in the next totally unrelated change....

File: oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java (hunk 2 above)
Comment (YoshiEnVerde) [reply oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java]: These are god points.
I'll add some better array length and string content checking, plus run the code through a code formatter.

Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java]: Other thing I've been thinking is that it may not be necessary to check for the ") " on the name.  We know there will be 52 fields, so if the split length is >52 we can subtract 52 to know how big the offset is (and how many fields to append to the name.)   This could wrap the length check in with a useful determination of nameOffset.  (And then you could piece together the name and strip off first and last characters in one go.)

**Comment by dbwiddis** [comment]
How's this PR coming? :)
